### PR TITLE
Write migration file to add missed optimism donations to db

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
This PR will add the missed donations that mentioned in https://github.com/Giveth/impact-graph/issues/1233 to DB

### CODE REVIEW

**Review by aminlatifi — CHANGES_REQUESTED** [review]
This donation is missed [0x8838d5be728ac29110340aacd883e9ede19048146e1c28842e4f4e28e4ba4761](https://optimistic.etherscan.io/tx/0x8838d5be728ac29110340aacd883e9ede19048146e1c28842e4f4e28e4ba4761)

File: migration/1703398409668-add_missed_op_donations_to_db.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+            "segmentNotified", "isTokenEligibleForGivback", "isProjectVerified", "createdAt", "givbackFactor", "powerRound", "projectRank", "bottomRankInRound", "qfRoundId", "tokenAddress")
+           VALUES ('${tx.toWalletAddress}', ${tx.projectId}, '${tx.fromWalletAddress}', ${user.id}, ${tx.amount}, '${tx.currency}', '${tx.transactionId}', ${tx.transactionNetworkId}, false, ${tx.valueUsd}, 'verified',
+             true, true, true, '${createdAt}', ${givbackFactor}, ${powerRound}, ${projectRank}, ${bottomRankInRound}, ${QF_ROUND_ID}, '${tx.tokenAddress}');
+                `);
+
+      await updateUserTotalDonated(user.id);
+      await updateUserTotalReceived(project.adminUser?.id);
+      await updateTotalDonationsOfProject(tx.projectId as number);
+      await refreshProjectEstimatedMatchingView();
+      await refreshProjectDonationSummaryView();
```
Comment (aminlatifi) [inline migration/1703398409668-add_missed_op_donations_to_db.ts]: These two can be done out of loop and only once

File: migration/1703398409668-add_missed_op_donations_to_db.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    currency: 'USDC',
+    tokenAddress: '0x7F5c764cBc14f9669B88837ca1490cCa17c31607',
+    amount: 3,
+    valueUsd: 3,
+    transactionNetworkId: NETWORK_IDS.OPTIMISTIC,
+    createdAt: millisecondTimestampToDate(1703166533000),
+  },
+
+  // https://optimistic.etherscan.io/tx/0x8838d5be728ac29110340aacd883e9ede19048146e1c28842e4f4e28e4ba4761
+  {
```
Comment (mohammadranjbarz) [inline migration/1703398409668-add_missed_op_donations_to_db.ts:104]: @aminlatifi  I added the missed dontion here

File: migration/1703398409668-add_missed_op_donations_to_db.ts [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        await calculateGivbackFactor(tx.projectId as number);
+
+      await queryRunner.query(`
+           INSERT INTO donation ("toWalletAddress", "projectId", "fromWalletAddress", "userId", amount, currency, "transactionId", "transactionNetworkId", anonymous, "valueUsd", status,
+            "segmentNotified", "isTokenEligibleForGivback", "isProjectVerified", "createdAt", "givbackFactor", "powerRound", "projectRank", "bottomRankInRound", "qfRoundId", "tokenAddress")
+           VALUES ('${tx.toWalletAddress?.toLowerCase()}', ${
+        tx.projectId
+      }, '${tx.fromWalletAddress?.toLocaleLowerCase()}', ${user.id}, ${
+        tx.amount
+      }, '${tx.currency}', '${tx.transactionId?.toLocaleLowerCase()}', ${
```
Comment (mohammadranjbarz) [inline migration/1703398409668-add_missed_op_donations_to_db.ts:267]: @aminlatifi  I made the addresses lowercase before inserting in DB

**Review by aminlatifi — CHANGES_REQUESTED** [review]

**Comment by aminlatifi** [comment]
The data is correct now, thanks @mohammadranjbarz 
I checked the transaction data and the integrity with project id

File: migration/1703398409668-add_missed_op_donations_to_db.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply migration/1703398409668-add_missed_op_donations_to_db.ts]: Thanks @aminlatifi , you are right, I moved it to out of the function

**Review by aminlatifi — APPROVED** [review]

**Comment by aminlatifi** [comment]
> The data is correct now, thanks @mohammadranjbarz I checked the transaction data and the integrity with project id

@CarlosQ96 Would you please carefully check the donation saving part and be sure everything is following the procedure?

**Review by CarlosQ96 — APPROVED** [review]
Checked the data inserted in the db, seems good, I added the missed donation Moe reported.

**Pull request merged**